android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java

```diff
@@ -235,5 +250,30 @@ public void onDestroyView() {
         if(viewPager != null && onPageChangeListener != null)
             viewPager.removeOnPageChangeListener(onPageChangeListener);
     }
+
+    @Override
+    public void showSnackbar(int toDisplayResource, int length, int actionCode, int sessionId, ImageButton slot_bookmark, int colorCode) {
+        Snackbar snackbar = Snackbar.make(coordinatorLayoutParent, toDisplayResource, length);
+        if (actionCode == CODE_UNDO_ADDED) {
+            snackbar = snackbar.setAction(R.string.undo, view -> {
+                RealmDataRepository.getDefaultInstance().setBookmark(sessionId, false).subscribe();
+                slot_bookmark.setImageResource(R.drawable.ic_bookmark_white_24dp);
+
+                WidgetUpdater.updateWidget(getContext());
+            })
+                    .setActionTextColor(ColorUtils.setAlphaComponent(Views.getAccentColor(getContext()), 220));
+        } else if (actionCode == CODE_UNDO_REMOVED) {
+            snackbar = snackbar.setAction(R.string.undo, view -> {
+                RealmDataRepository.getDefaultInstance().setBookmark(sessionId, true).subscribe();
+                slot_bookmark.setImageResource(R.drawable.ic_bookmark_white_24dp);
+                slot_bookmark.setColorFilter(colorCode, PorterDuff.Mode.SRC_ATOP);
+
+                WidgetUpdater.updateWidget(getContext());
+            })
+                    .setActionTextColor(ColorUtils.setAlphaComponent(Views.getAccentColor(getContext()), 220));
+        }
+
+        snackbar.show();
+    }
```

Comment:
This code is almost same everywhere, shouldn't this be handled in just one place, the activity?

Reply:
Sorry didn't get you. Do you mean like a helper function for making the Snackbar?

Reply:
I mean if there is no change, and it only occurs in a single fragment hierarchy, why can't it be handled in a single place

Reply:
We need to pass the parentView in each fragment while making the Snackbar. So I can make a helper method where I pass the view from the fragments and the rest of the code is the same everywhere. Didnt know if this was good practice or not. Should I do it like this?

Reply:
Yup, if the code depends on each fragment's layout, then create a helper method for all the similar code